### PR TITLE
Maestro config changes for dell 5000

### PR DESCRIPTION
**Pull request by SenRamakri**
This PR contains previous changes from @costanic @mray190 and my latest changes to fix many issues causing maestro crashing on Dell 5000.

### CODE REVIEW

**Comment by SenRamakri** [comment]
@x85446 - Please review and merge the maestro config changes for Dell 5000.

File: conf/maestro-conf/edge-config-dell5000-demo.yaml [hunk 1]
```diff
@@ -29,34 +31,38 @@ platform_readers:
 #     This is only a valid option if you are not using a well-known CA
 #    server_name: "devcloud.wigwag.io"
 var_defs:
+   - key: "SNAP_DATA"
+     value: "/var/snap/pelion-edge/current"
+   - key: "SNAP"
+     value: "/snap/pelion-edge/current"
```
Comment (cmonr) [inline conf/maestro-conf/edge-config-dell5000-demo.yaml]: Making sure, does this value also need a `/var` prefix?

File: conf/maestro-conf/edge-config-dell5000-demo.yaml [hunk 2]
```diff
@@ -29,34 +31,38 @@ platform_readers:
...
+     value: "{{SNAP}}/bin/node"
    - key: "DEVICEJS_ROOT"
-     value: "/snap/pelion-edge/current/wigwag/devicejs-ng"
+     value: "{{SNAP}}/wigwag/devicejs-ng"
    - key: "DEVJS_CORE_MODULES"
-     value: "/snap/pelion-edge/current/wigwag/devicejs-core-modules"
+     value: "{{SNAP}}/wigwag/devicejs-core-modules"
    - key: "MAESTRO_RUNNER_DIR"
-     value: "/snap/pelion-edge/current/wigwag/devicejs-core-modules/maestroRunner"
+     value: "{{SNAP}}/wigwag/devicejs-core-modules/maestroRunner"
```
Comment (cmonr) [inline conf/maestro-conf/edge-config-dell5000-demo.yaml]: How do you know which variables need `{{SNAP}}`, and which need `{{SNAP_VAR}}`?

File: conf/maestro-conf/edge-config-dell5000-demo.yaml [hunk 3]
```diff
@@ -29,34 +31,38 @@ platform_readers:
...
      value: 9000
    - key: "LOCAL_DATABASE_STORAGE_DIRECTORY"
-     value: "/snap/pelion-edge/current/userdata/etc/devicejs/db"
+     value: "{{SNAP_DATA}}/userdata/etc/devicejs/db"
    - key: "RELAY_VERSIONS_FILE"
-     value: "/snap/pelion-edge/current/wigwag/etc/versions.json"
+     value: "{{SNAP}}/wigwag/etc/versions.json"
    - key: "FACTORY_VERSIONS_FILE"
-     value: "/mnt/.overlay/factory/wigwag/etc/versions.json"
+     value: "/snap/pelion-edge/x1/wigwag/etc/versions.json"
```
Comment (cmonr) [inline conf/maestro-conf/edge-config-dell5000-demo.yaml]: What is `x1`?

File: conf/maestro-conf/edge-config-dell5000-demo.yaml (hunk 2 above)
Comment (SenRamakri) [reply conf/maestro-conf/edge-config-dell5000-demo.yaml]: Good question - The SNAP_VAR is writeable and SNAP is not. So, whenever we need to write something out we point it to SNAP_VAR.

File: conf/maestro-conf/edge-config-dell5000-demo.yaml (hunk 3 above)
Comment (SenRamakri) [reply conf/maestro-conf/edge-config-dell5000-demo.yaml]: I think it refers to the version we want to use as snap is capable of creating multiple versions if we need to rollback. Currently we only have have version 1.

File: conf/maestro-conf/edge-config-dell5000-demo.yaml (hunk 1 above)
Comment (SenRamakri) [reply conf/maestro-conf/edge-config-dell5000-demo.yaml]: I think that looks correct, SNAP should point to /snap which is the read-only partition.

File: conf/maestro-conf/edge-config-dell5000-demo.yaml (hunk 2 above)
Comment (cmonr) [reply conf/maestro-conf/edge-config-dell5000-demo.yaml]: > The SNAP_VAR is writeable 

Slightly different wording, to make sure I understand.

The `SNAP_VAR` _directory_ is rw.
The `SNAP` _directory_ is ro.

Correct?

Comment (SenRamakri) [reply conf/maestro-conf/edge-config-dell5000-demo.yaml]: That's correct, to be accurate its SNAP and SNAP_DATA. SNAP_DATA(which actually points to /var/snap) is rw and SNAP is ro.

**Pull request merged**